ExaConfig parsing fails when root pasword was set

Inside the docker db you can set the root password using:

```shell
exaconf passwd-user -n root
exaconf commit
```

This adds a line like the following to `/exa/etc/EXAConf`:

```
Passwd = $6$47bbYJHqGkMVR6f8$fCscTE1gcz6G6aXyojDwGGFe4qPYsIfmzqPr2Q.vZLpU/XADzKECdzCHezPsmlUnSSzzwSBXkMEh6E1pQoKO8/
```
This line breaks, however, the parsing of the ExaConf:

```
org.testcontainers.containers.ContainerLaunchException: Container startup failed
...
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:323)
	... 73 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:497)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:325)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	... 74 more
Caused by: java.lang.IllegalArgumentException: Illegal base64 character 24
	at java.base/java.util.Base64$Decoder.decode0(Base64.java:746)
	at java.base/java.util.Base64$Decoder.decode(Base64.java:538)
	at java.base/java.util.Base64$Decoder.decode(Base64.java:561)
	at com.exasol.exaconf.ConfigurationParser.decodePassword(ConfigurationParser.java:68)
	at com.exasol.exaconf.ConfigurationParser.parseUnindentedLine(ConfigurationParser.java:57)
	at com.exasol.exaconf.ConfigurationParser.parseLine(ConfigurationParser.java:41)
	at com.exasol.exaconf.ConfigurationParser.parse(ConfigurationParser.java:33)
	at com.exasol.containers.ExasolContainer.readClusterConfiguration(ExasolContainer.java:505)
	at com.exasol.containers.ExasolContainer.clusterConfigurationIsAvailable(ExasolContainer.java:491)
	at com.exasol.containers.ExasolContainer.waitUntilClusterConfigurationAvailable(ExasolContainer.java:487)
	at com.exasol.containers.ExasolContainer.waitUntilContainerStarted(ExasolContainer.java:396)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:440)
```

--- a/doc/changes/changelog.md
+++ b/doc/changes/changelog.md
@@ -1,5 +1,6 @@
 # Changes
 
+* [3.5.3](changes_3.5.3.md)
 * [3.5.2](changes_3.5.2.md)
 * [3.5.1](changes_3.5.1.md)
 * [3.5.0](changes_3.5.0.md)

--- a/doc/changes/changes_3.5.3.md
+++ b/doc/changes/changes_3.5.3.md
@@ -1,0 +1,13 @@
+# Exasol Test Containers 3.5.3, released 2021-05-14
+
+Code name: Fixed EXAConf parsing
+
+## Summary
+
+In this release we fixed a bug that lead to a crash when the root password was set in the EXAConf.
+
+## Bugfixes
+
+* #136: Fixed parsing of EXAConfig with root password set
+
+## Dependency Updates

--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.exasol</groupId>
     <artifactId>exasol-testcontainers</artifactId>
-    <version>3.5.2</version>
+    <version>3.5.3</version>
     <name>Test containers for Exasol on Docker</name>
     <description>This module provides abstraction for generation, startup, shutdown and use of an
         Exasol database running on Docker.

--- a/release_config.yml
+++ b/release_config.yml
@@ -1,0 +1,7 @@
+community-tags:
+  - docker
+  - docker-db
+  - integration-testing
+community-project-name: Exasol Testcontainers
+community-project-description: |
+  Exasol test containers provides an abstraction to Exasol running on Docker for the purpose of integration testing based on Java Testcontainers.

--- a/src/main/java/com/exasol/exaconf/ConfigurationParser.java
+++ b/src/main/java/com/exasol/exaconf/ConfigurationParser.java
@@ -12,6 +12,7 @@ public class ConfigurationParser {
     private String subsection = "";
     private final Map<String, String> parameters = new HashMap<>();
     private final String rawConfig;
+    private static final Set<String> BASE64_ENCODED_VALUES = Set.of("ReadPasswd", "WritePasswd");
 
     /**
      * Create a new instance of a {@link ConfigurationParser}.
@@ -28,7 +29,7 @@ public class ConfigurationParser {
      * @return Configuration as object
      */
     public ClusterConfiguration parse() {
-        try (Scanner scanner = new Scanner(this.rawConfig)) {
+        try (final Scanner scanner = new Scanner(this.rawConfig)) {
             while (scanner.hasNext()) {
                 parseLine(scanner.nextLine());
             }
@@ -54,7 +55,7 @@ public class ConfigurationParser {
             if (assignmentOperatorPosition > 0) {
                 final String key = line.substring(0, assignmentOperatorPosition - 1).trim();
                 final String value = line.substring(assignmentOperatorPosition + 1).trim();
-                final String decodedValue = (key.endsWith("Passwd") ? decodePassword(value) : value);
+                final String decodedValue = (BASE64_ENCODED_VALUES.contains(key) ? decodePassword(value) : value);
                 this.parameters.put(this.section + this.subsection + key, decodedValue);
             }
         }

--- a/src/test/java/com/exasol/exaconf/ConfigurationParserTest.java
+++ b/src/test/java/com/exasol/exaconf/ConfigurationParserTest.java
@@ -3,6 +3,7 @@ package com.exasol.exaconf;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.*;
 import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
 
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
@@ -110,5 +111,13 @@ class ConfigurationParserTest {
         final ClusterConfiguration clusterConfiguration = parseConfiguration(
                 "[BucketFS : Fred]\nHttpPort=1\n[[Bucket : default]]\n[BucketFS : Wilma]\nHttpPort=2\n[BucketFS : Barney]\nHttpPort=3");
         assertThat(clusterConfiguration.getBucketFsServiceNames(), contains("Fred", "Wilma", "Barney"));
+    }
+
+    @Test
+    void testParseConfigurationWithRootPasswordSet() {
+        assertDoesNotThrow(() -> parseConfiguration("    [[root]]\n" + "        ID = 0\n" + "        Group = root\n"
+                + "        LoginEnabled = True\n"
+                + "        AdditionalGroups = exausers, exadbadm, exastoradm, exabfsadm, exaadm\n"
+                + "        Passwd = $6$ypRK9ia5lkj9/DWF$SxzzizsUp4AZYmk2m2PgKeA8fDT4Ou3FreNQkPFBoTbaR4HI0gNs0o4lEwtAAManFUeU00iq8c/mfAWmtyLQI/\n"));
     }
 }

